flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolImplTest.java

```diff
@@ -738,6 +740,95 @@ public void testCalculationOfTaskExecutorUtilization() throws Exception {
 		}
 	}
 
+	@Test
+	public void testOrphanedAllocationCanBeRemapped() throws Exception {
+		try (SlotPoolImpl slotPool = createSlotPoolImpl()) {
+			final List<AllocationID> allocationIds = new ArrayList<>();
+			resourceManagerGateway.setRequestSlotConsumer(
+				slotRequest -> allocationIds.add(slotRequest.getAllocationId()));
+
+			final List<AllocationID> canceledAllocations = new ArrayList<>();
+			resourceManagerGateway.setCancelSlotConsumer(canceledAllocations::add);
```

Comment:
nit: maybe, not now but if it can be reused also in other tests, it would be nice to have something like an RM harness:
```
class RmHarness {
final List<AllocationID> allocationIds = new ArrayList<>();
final List<AllocationID> canceledAllocations = new ArrayList<>();
RmHarness(resourceManagerGateway)
getAllocations
getCanceled
}
```

Reply:
it can be partially reused in `testFailingAllocationFailsRemappedPendingSlotRequests`

Reply:
Maybe later along with the rework of all SlotPool tests? Because resourceManagerGateway is shared between all cases and it's better to have an overview of the usages.